src/engine/platform/mixer.h

```diff
@@ -32,39 +32,23 @@ struct AudioID {
     u16 slot;
 };
 
-struct EffectID {
-    s32 channel;
-    u32 slot;
-    u32 gen;
-
-    bool operator== (const EffectID &other) const {
-        return channel == other.channel && slot == other.slot && gen == other.gen;
-    }
-
-    operator bool() const {
-        return channel >= 0;
-    }
+struct Channel {
+    f32 *buffer;
+
+    struct {
+        bool active;
+        f32 feedback;
```

Comment:
Doesn't the feedback field implicitly work like an active flag? I don't have strong feelings here but you don't need to bool field.

Reply:
I switched to overloading the bool-operator instead.